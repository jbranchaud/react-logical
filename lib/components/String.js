import React from 'react';
import PropTypes from 'prop-types';

const String = ({ str, blankStr }) => {
  if (str && str.length) {
    return (
      <React.Fragment>
        {str}
      </React.Fragment>
    );
  } else {
    return (
      <React.Fragment>
        {blankStr}
      </React.Fragment>
    );
  }
};

String.defaultProps = {
  str: '',
  blankStr: '',
};

String.propTypes = {
  str: PropTypes.string,
  blankStr: PropTypes.string,
};

export default String;
