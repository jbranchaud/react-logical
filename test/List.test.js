import React from 'react';
import renderer from 'react-test-renderer';

import String from '../lib/components/String';

it('renders with no props', () => {
  const tree = renderer.create(<String />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with a string', () => {
  const tree = renderer.create(<String str={'Logical'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with a blank string', () => {
  const tree = renderer
    .create(<String str={''} blankStr={'Fallthrough'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with no string', () => {
  const tree = renderer.create(<String blankStr={'Fallthrough'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
