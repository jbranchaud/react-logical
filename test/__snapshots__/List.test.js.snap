// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a blank string 1`] = `"Fallthrough"`;

exports[`renders with a string 1`] = `"Logical"`;

exports[`renders with no props 1`] = `""`;

exports[`renders with no string 1`] = `"Fallthrough"`;
